cpp/cgi_progs/zotero_cgi.cc

```diff
@@ -406,6 +409,101 @@ void ProcessShowDownloadedAction(const std::multimap<std::string, std::string> &
 }
 
 
+void ProcessShowQAAction(const std::multimap<std::string, std::string> &cgi_args) {
+    const std::string zeder_id(GetCGIParameterOrDefault(cgi_args, "zeder_id"));
+    const std::string zeder_instance(GetCGIParameterOrDefault(cgi_args, "zeder_instance"));
+    std::string zeder_journal_id;
+    std::string journal_name;
+    std::string submitted("false");
+
+    // try to get more details for given journal
+    DbConnection db_connection;
+    {
+        db_connection.query("SELECT id,journal_name FROM zeder_journals WHERE zeder_id=" + db_connection.escapeAndQuoteString(zeder_id) +
+                            " AND zeder_instance=" + db_connection.escapeAndQuoteString(zeder_instance));
+        auto result_set(db_connection.getLastResultSet());
+        while (auto row = result_set.getNextRow()) {
+            zeder_journal_id = row["id"];
+            journal_name = row["journal_name"];
+        }
+    }
+
+    // sub-action "delete"
```

Comment:
I think the sub-actions should be implemented in individual functions.

Reply:
yeah, right (already pondered about whether this would be good or not => it is now, less spaghetti)

done